src/librustc/ty/sty.rs

```diff
@@ -2154,30 +2154,26 @@ impl<'tcx> Const<'tcx> {
     pub fn try_eval_bits(
         &self,
         tcx: TyCtxt<'_, '_, '_>,
-        ty: ParamEnvAnd<'_, Ty<'_>>,
+        param_env: ParamEnv<'_>,
+        ty: Ty<'_>,
     ) -> Option<u128> {
-        assert_eq!(self.ty, ty.value);
+        assert_eq!(self.ty, ty);
         let tcx = tcx.global_tcx();
-        let ty = if ty.value.needs_infer() || ty.value.has_placeholders() {
-            tcx
-                .lift_to_global(&ty)
-                .unwrap_or_else(|| bug!("cannot lift {:#?} for {:#?}", ty, self))
-        } else {
-            let ty = tcx
-                .lift_to_global(&ty.value)
-                .unwrap_or_else(|| bug!("cannot lift {:#?} for {:#?}", ty, self));
-            ParamEnv::empty().and(ty)
-        };
+        // if `ty` has no unresolved components, use an empty param_env
```

Comment:
I don't think this comment is accurate.
But I'd defer to @nikomatsakis. What I think ParamEnv::add does today is throw away the bounds if the value doesn't refer to any of the generic parameters in scope.
So less unresolved and more "generic".

Reply:
Yes that looks like what the source does and makes much more sense. I'll adjust the comment accordingly